internal/service/alertconfiguration/model_alert_configuration.go

```diff
@@ -0,0 +1,333 @@
+		newState.VictorOpsAPIKey = conversion.StringNullIfEmpty(currState.VictorOpsAPIKey.ValueString())
+		newState.VictorOpsRoutingKey = conversion.StringNullIfEmpty(currState.VictorOpsRoutingKey.ValueString())
+		newState.WebhookURL = conversion.StringNullIfEmpty(currState.WebhookURL.ValueString())
+		newState.WebhookSecret = conversion.StringNullIfEmpty(currState.WebhookSecret.ValueString())
+		newState.MicrosoftTeamsWebhookURL = conversion.StringNullIfEmpty(currState.MicrosoftTeamsWebhookURL.ValueString())
+
+		// for optional attributes that are not computed we must check if they were previously defined in state
+		if !currState.ChannelName.IsNull() {
+			newState.ChannelName = conversion.StringPtrNullIfEmpty(value.ChannelName)
+		}
+		if !currState.DatadogRegion.IsNull() {
+			newState.DatadogRegion = conversion.StringPtrNullIfEmpty(value.DatadogRegion)
+		}
+		if !currState.EmailAddress.IsNull() {
+			newState.EmailAddress = conversion.StringPtrNullIfEmpty(value.EmailAddress)
+		}
+		if !currState.MobileNumber.IsNull() {
+			newState.MobileNumber = conversion.StringPtrNullIfEmpty(value.MobileNumber)
+		}
+		if !currState.OpsGenieRegion.IsNull() {
+			newState.OpsGenieRegion = conversion.StringPtrNullIfEmpty(value.OpsGenieRegion)
+		}
+		if !currState.TeamID.IsNull() {
+			newState.TeamID = conversion.StringPtrNullIfEmpty(value.TeamId)
+		}
+		if !currState.TypeName.IsNull() {
+			newState.TypeName = conversion.StringPtrNullIfEmpty(value.TypeName)
+		}
+		if !currState.Username.IsNull() {
+			newState.Username = conversion.StringPtrNullIfEmpty(value.Username)
+		}
+
+		newState.NotifierID = types.StringPointerValue(value.NotifierId)
+		newState.IntervalMin = types.Int64PointerValue(conversion.IntPtrToInt64Ptr(value.IntervalMin))
+		newState.DelayMin = types.Int64PointerValue(conversion.IntPtrToInt64Ptr(value.DelayMin))
+		newState.EmailEnabled = types.BoolValue(value.EmailEnabled != nil && *value.EmailEnabled)
+		newState.SMSEnabled = types.BoolValue(value.SmsEnabled != nil && *value.SmsEnabled)
+
+		notifications[i] = newState
+	}
+
+	return notifications
+}
+
+func NewTFMetricThresholdConfigModel(t *admin.ServerlessMetricThreshold, currStateSlice []TfMetricThresholdConfigModel) []TfMetricThresholdConfigModel {
+	if t == nil {
+		return []TfMetricThresholdConfigModel{}
+	}
+	if len(currStateSlice) == 0 { // metric threshold was created elsewhere from terraform, or import statement is being called
+		return []TfMetricThresholdConfigModel{
+			{
+				MetricName: conversion.StringNullIfEmpty(t.MetricName),
+				Operator:   conversion.StringNullIfEmpty(*t.Operator),
+				Threshold:  types.Float64Value(*t.Threshold),
+				Units:      conversion.StringNullIfEmpty(*t.Units),
+				Mode:       conversion.StringNullIfEmpty(*t.Mode),
+			},
+		}
+	}
+	currState := currStateSlice[0]
+	newState := TfMetricThresholdConfigModel{}
+	if !currState.MetricName.IsNull() {
+		newState.MetricName = conversion.StringNullIfEmpty(t.MetricName)
```

Comment:
same, no need to do if, we can just do: 

newState := TfMetricThresholdConfigModel{ .... values }

Reply:
same response as previous comments